internal/http/parser/http_response.go

```diff
@@ -0,0 +1,225 @@
+package http_parser
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type Header struct {
+	Key   string
+	Value string
+}
+
+type StatusLine struct {
+	Version    string
+	StatusCode int
+	Reason     string
+}
+
+type HTTPResponse struct {
+	StatusLine StatusLine
+
+	Headers []Header
+
+	Body []byte
+}
+
+func Parse(data []byte) (httpResponse HTTPResponse, readBytesCount int, err error) {
+	reader := bytes.NewReader(data)
+
+	response, err := doParseResponse(reader)
+	if err != nil {
+		return HTTPResponse{}, 0, err
+	}
+
+	return response, len(data) - reader.Len(), nil
+}
+
+func doParseResponse(reader *bytes.Reader) (HTTPResponse, error) {
+	var allHeadersFound = false
+	var sectionsFound int
```

Comment:
The name here sounds like a boolean, better name would be `sectionsCount`

Reply:
Also I think the more canonical way to write this is to just do assignments: 

sectionsCount := 0